core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostBasedScheduleGenerator.scala

```diff
@@ -601,16 +589,31 @@ class CostBasedScheduleGenerator(
   }
 
   /**
-    * The cost function used by the search. Takes a region plan, generates one or more (to be done in the future)
-    * schedules based on the region plan, and calculates the cost of the schedule(s) using Cost Estimator. Uses the cost
-    * of the best schedule (currently only considers one schedule) as the cost of the region plan.
+    * Takes a region DAG, generates one or more (to be done in the future) schedules based on the region DAG, allocates
+    * resources to each region in the region DAG, and calculates the cost of the schedule(s) using Cost Estimator. Uses
+    * the cost of the best schedule (currently only considers one schedule) as the cost of the region DAG.
     *
     * @return A cost determined by the cost estimator.
     */
-  private def evaluate(regionPlan: RegionPlan): Double = {
+  private def allocateResourcesAndEvaluateCost(
+      regionDAG: DirectedAcyclicGraph[Region, RegionLink]
+  ): Double = {
+    val regionPlan =
+      RegionPlan(regionDAG.vertexSet().asScala.toSet, regionDAG.edgeSet().asScala.toSet)
     val schedule = generateScheduleFromRegionPlan(regionPlan)
     // In the future we may allow multiple regions in a level and split the resources.
-    schedule.map(level => level.map(region => costEstimator.estimate(region, 1)).sum).sum
+    schedule
+      .map(level =>
+        level
+          .map(region => {
+            val (newRegion, regionCost) = costEstimator.allocateResourcesAndEstimateCost(region, 1)
```

Comment:
`newRegion` is confusing. Better to return a resource config, instead of the region with the updated resource.

Reply:
Done